### ISSUE TITLE
E2520 raised for mutually exclusive properties when using Conditions

### ISSUE THREAD

**Issue by samoznak** [aws-cloudformation/cfn-lint#2885]
### CloudFormation Lint Version

cfn-lint 0.80.2

### What operating system are you using?

Windows

### Describe the bug

[E2520](https://github.com/aws-cloudformation/cfn-lint/blob/main/docs/rules.md#E2520) is raised for mutually exclusive properties when using Conditions

```
cfn-lint -t ./template.yaml
E2520 Property SourceSecurityGroupId should NOT exist with CidrIp for Resources/Ingress/Properties
.\template.yaml:13:7
```

The same was working prior `0.79.11`. PR  [2875](https://github.com/aws-cloudformation/cfn-lint/pull/2875) seems to be the cause.

```
> cfn-lint --version            
cfn-lint 0.79.10
> cfn-lint -t ./template.yaml   
> echo $lastexitcode
0
```

### Expected behavior

E2520 is ignored for mutually exclusive properties that use the same Condition and Fn::If intrinsic function which makes sure only one of the properties has value.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Parameters:
  pCidr:
    Type: String
    Default: ''
Conditions:
  cIsCidr: !Not [!Equals [!Ref pCidr, '']]
Resources:
  Ingress:
    Type: AWS::EC2::SecurityGroupIngress
    Properties:
      SourceSecurityGroupId: !If [ cIsCidr, !Ref AWS::NoValue, sg-abc12345 ]
      CidrIp: !If [ cIsCidr, !Ref pCidr, !Ref AWS::NoValue ]
      IpProtocol: "-1"
      GroupId: sg-abc1234567
```

**Comment by samoznak** [comment]
I can confirm [0.80.3](https://github.com/aws-cloudformation/cfn-lint/releases/tag/v0.80.3) fixes the issue for me. Thank you @kddejong for quick fix and release!
